api/swagger.yml

```diff
@@ -1,507 +1,795 @@
-        enum: [ failed, completed ]
-      commit_id:
-        type: string
+    ActionRunList:
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/ActionRun"
+
+    HookRun:
+      type: object
+      required:
+        - hook_run_id
+        - action
+        - hook_id
+        - start_time
+        - status
+      properties:
+        hook_run_id:
+          type: string
+        action:
+          type: string
+        hook_id:
+          type: string
+        start_time:
+          type: string
+          format: date-time
+        end_time:
+          type: string
+          format: date-time
+        status:
+          type: string
+          enum: [ failed, completed ]
+
+    HookRunList:
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/HookRun"
+
+    StagingLocation:
+      type: object
+      description: location for placing an object when staging it
+      required:
+        - token
+      properties:
+        physical_address:
+          type: string
+        token:
+          type: string
+          description: opaque staging token to use to link uploaded object
+
+    StagingMetadata:
+      type: object
+      description: information about uploaded object
+      properties:
+        staging:
+          $ref: "#/components/schemas/StagingLocation"
+        checksum:
+          type: string
+          description: unique identifier of object content on backing store (typically ETag)
+        mtime:
+          type: integer
+          format: int64
+        size_bytes:
+          type: integer
+          format: int64
+      required:
+        - staging
+        - checksum
+        - mtime
+        - size_bytes
 
-  hook_run:
-    type: object
-    required:
-      - hook_run_id
-    properties:
-      hook_run_id:
-        type: string
-      action:
-        type: string
-      hook_id:
-        type: string
-      start_time:
-        type: string
-        format: date-time
-      end_time:
-        type: string
-        format: date-time
-      status:
-        type: string
-        enum: [ failed, completed ]
 
 paths:
 
   /setup_lakefs:
     post:
       tags:
-        - setup
-      operationId: setupLakeFS
-      summary: setup lakeFS and create the first user
+        - config
+      operationId: setup
+      summary: setup lakeFS and create a first user
       security: [ ]
-      parameters:
-        - in: body
-          name: user
-          schema:
-            $ref: "#/definitions/setup"
+      requestBody:
+        required: true
+        content:
+          application/json:
```

Comment:
This will get annoying fairly quickly.  Can we somehow default this away, like we could with Swagger?

Reply:
sorry, didn't found a way in the spec to default this one.